src/lib.rs

```diff
@@ -162,31 +168,22 @@ impl Log for AndroidLogger {
             return;
         }
 
-        // tag longer than LOGGING_TAG_MAX_LEN causes allocation
+        // Temporary storage for null-terminating record.module_path() if it's needed.
+        // Tags too long to fit here cause allocation.
         let mut tag_bytes: [MaybeUninit<u8>; LOGGING_TAG_MAX_LEN + 1] = uninit_array();
+        // In case we end up allocating, keep the CString alive.
+        let _owned_tag;
 
         let module_path = record.module_path().unwrap_or_default();
 
-        // If no tag was specified, use module name
-        let custom_tag = &config.tag;
-        let tag = custom_tag
-            .as_ref()
-            .map(|s| s.as_bytes())
-            .unwrap_or_else(|| module_path.as_bytes());
-
-        // In case we end up allocating, keep the CString alive.
-        let _owned_tag;
-        let tag = if tag.len() < tag_bytes.len() {
-            // truncate the tag here to fit into LOGGING_TAG_MAX_LEN
-            fill_tag_bytes(&mut tag_bytes, tag)
+        let tag: &CStr = if let Some(ref tag) = config.tag {
```

Comment:
I removed this in a previous PR, it doesn't seem like we need a type annotation here.

(Also instead of `ref` it seems people more commonly borrow the right-hand-side statement instead)

```suggestion
        let tag = if let Some(tag) = &config.tag {
```

Reply:
Fixed now :)